contracts/beyondFramework/dca/base/DCABaseUpgradeable.sol

```diff
@@ -0,0 +1,1168 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import { InvestQueueLib } from "../libraries/InvestQueueLib.sol";
+import { DcaHistoryLib } from "../libraries/DcaHistoryLib.sol";
+import { IDcaStrategy } from "../interfaces/IDcaStrategy.sol";
+import { SwapLib } from "../libraries/SwapLib.sol";
+
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/interfaces/IERC20Upgradeable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { PausableUpgradeable } from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+abstract contract DCABaseUpgradeable is
+    OwnableUpgradeable,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    IDcaStrategy
+{
+    using InvestQueueLib for InvestQueueLib.InvestQueue;
+    using DcaHistoryLib for DcaHistoryLib.DcaHistory;
+    using SwapLib for SwapLib.Router;
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    DepositFee public depositFee;
+
+    address public dcaInvestor;
+    address[] public portfolios;
```

Comment:
minor: first I didn't understand why we have portfolios in a strategy, but I realized it is the whitelistedPortfolios, should we change the name?